cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -875,3 +496,412 @@ def _act_on_(self, args: 'cirq.ActOnArgs', qubits: Sequence['cirq.Qid']) -> bool
+            * {middle point of xyz in 4 Quadrant} * 120 is [[0, 1], [1, 1]]
+            * {middle point of xyz in 4 Quadrant} * 240 is [[1, 1], [1, 0]]
+        """
+        x_to_flip, z_to_flip = self.clifford_tableau.rs
+        flip_index = int(z_to_flip) * 2 + x_to_flip
+        a, x, z = 0.0, 0.0, 0.0
+
+        if np.array_equal(self.clifford_tableau.matrix(), [[1, 0], [0, 1]]):
+            # I, Z, X, Y cases
+            to_phased_xz = [(0.0, 0.0, 0.0), (0.0, 0.0, 1.0), (0.0, 1.0, 0.0), (0.5, 1.0, 0.0)]
+            a, x, z = to_phased_xz[flip_index]
+        elif np.array_equal(self.clifford_tableau.matrix(), [[1, 0], [1, 1]]):
+            # +/- X_sqrt, 2 Hadamard-like gates acting on the YZ plane
+            a = 0.0
+            x = 0.5 if x_to_flip ^ z_to_flip else -0.5
+            z = 1.0 if x_to_flip else 0.0
+        elif np.array_equal(self.clifford_tableau.matrix(), [[0, 1], [1, 0]]):
+            # +/- Y_sqrt, 2 Hadamard-like gates acting on the XZ plane
+            a = 0.5
+            x = 0.5 if x_to_flip else -0.5
+            z = 0.0 if x_to_flip ^ z_to_flip else 1.0
+        elif np.array_equal(self.clifford_tableau.matrix(), [[1, 1], [0, 1]]):
+            # +/- Z_sqrt, 2 Hadamard-like gates acting on the XY plane
+            to_phased_xz = [(0.0, 0.0, 0.5), (0.0, 0.0, -0.5), (0.25, 1.0, 0.0), (-0.25, 1.0, 0.0)]
+            a, x, z = to_phased_xz[flip_index]
+        elif np.array_equal(self.clifford_tableau.matrix(), [[0, 1], [1, 1]]):
+            # axis swapping rotation -- (312) permutation
+            a = 0.5
+            x = 0.5 if x_to_flip else -0.5
+            z = 0.5 if x_to_flip ^ z_to_flip else -0.5
+        else:
+            # axis swapping rotation -- (231) permutation.
+            # This should be the only cases left.
+            assert np.array_equal(self.clifford_tableau.matrix(), [[1, 1], [1, 0]])
+            a = 0.0
+            x = -0.5 if x_to_flip ^ z_to_flip else 0.5
+            z = -0.5 if x_to_flip else 0.5
+
+        return phased_x_z_gate.PhasedXZGate(x_exponent=x, z_exponent=z, axis_phase_exponent=a)
+
+    def __pow__(self, exponent) -> 'SingleQubitCliffordGate':
+        # First to check if we can get the sqrt and negative sqrt Clifford.
+        if self._get_sqrt_map().get(exponent, None):
+            pow_gate = self._get_sqrt_map()[exponent].get(self, None)
+            if pow_gate:
+                return pow_gate
+        # If not, we try the Clifford Tableau based method.
+        ret_gate = super().__pow__(exponent)
+        if ret_gate is NotImplemented:
+            return NotImplemented
+        return SingleQubitCliffordGate.from_clifford_tableau(ret_gate.clifford_tableau)
+
+    def _act_on_(
+        self,
+        args: 'cirq.OperationTarget',  # pylint: disable=unused-argument
+        qubits: Sequence['cirq.Qid'],  # pylint: disable=unused-argument
+    ):
+        # TODO(PR number) will create a PR if we agree on adding this.
```

Comment:
@daxfohl @MichaelBroughton Do you have any objection on this?

Reply:
No, but thanks for asking. I noticed this a few weeks ago and my initial thought was to ask for a change here. I'd been trying to remove all the isinstance checks from act_on because it made more sense that the simulator should know about gates but not vice versa. However here, I feel like this is an example of a gate that operates on a higher level than the simulator. So I think this is an example where it's okay.

Reply:
None from me. Can we resolve the merge conflicts so we can retry the auto merge ?

Reply:
Done